crates/subspace-networking/src/peer_info.rs

```diff
@@ -150,6 +206,15 @@ impl<PIP: PeerInfoProvider> NetworkBehaviour for Behaviour<PIP> {
         _: &mut Context<'_>,
         _: &mut impl PollParameters,
     ) -> Poll<ToSwarm<Self::OutEvent, THandlerInEvent<Self>>> {
+        if self.should_notify_handlers.swap(false, Ordering::SeqCst) {
+            debug!("Notify peer-info handlers.");
+
+            self.requests.clear();
+            for peer_id in self.connected_peers.iter().cloned() {
+                self.requests.push(Request { peer_id })
+            }
+        }
```

Comment:
This doesn't seem quite right either. You check it once, but what if value of `should_notify_handlers` changes to `true`, without waker stored no one will notify futures executor to poll the behavior again. you need to store waker internally and wake it once value changes.

Also `requests` looks more like `messages` or `notifications` (but it would overlap with other notifications) to me. You're not really requesting anything here.